bugsnag-android-core/src/test/java/com/bugsnag/android/EventDeliveryStrategyTest.kt

```diff
@@ -0,0 +1,99 @@
+package com.bugsnag.android
+
+import com.bugsnag.android.BugsnagTestUtils.generateImmutableConfig
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.junit.MockitoJUnitRunner
+import java.util.Date
+
+@RunWith(MockitoJUnitRunner::class)
+internal class EventDeliveryStrategyTest {
+
+    private val apiKey = "BUGSNAG_API_KEY"
+    private val notifier = Notifier()
+    private val config = generateImmutableConfig()
+    private var testException: RuntimeException? = null
+    private val event = Event(
+        RuntimeException("Whoops!"), config,
+        SeverityReason.newInstance(
+            SeverityReason.REASON_UNHANDLED_EXCEPTION
+        ),
+        NoopLogger
+    )
```

Comment:
Given that this is only used in `testGenerateUnhandledReport` it should be moved there